### PR TITLE
eq-1044 Submission button

### PR DESCRIPTION
**Pull request by hamishtaplin**
As a user, I need the action of submitting my answers to be clear, reliable and resilient so that I am confident my answers are being submitted and I am prevented from breaking the process.

# Acceptance Criteria

- User is given an indication when submitting their answers that something is happening
- User is prevented (as much as possible) from accidentally double-submitting/causing a UX failure
- Solution can be a progressive enhancement (not available to all users).
- Solution extensively tested on all browsers and devices, this could break submission.

### CODE REVIEW

**Comment by armstrongb** [comment]
Can we change this so that the spinner doesn't replace the copy entirely but is shown alongside copy: "[spinner] Submitting..."?
submission-spinner

**Comment by yoganandkunche** [comment]
@hamishtaplin: as discussed please add the spinner 'submitting' to MWSS, UKIS and Census submission screens.

**Comment by yoganandkunche** [comment]
@hamishtaplin timeout pop up displays '**Continuing**..' PFA -- Bug introduced on this branch.

screen shot 2017-04-12 at 16 32 37

**Comment by yoganandkunche** [comment]
@collisdigital I had a discussion with @hamishtaplin and we want you to take a decision before we go ahead with this feature... as at the moment the "[spinner] Submitting..." thing comes up for 5 secs only and the scenario where it is misleading to the respondent is when the network is bad and it takes more than 5 secs before it get submitted the [Submit survey] button is displayed again which might confuse the respondent to click it again which results in 'Session timeout'.

**Comment by collisdigital** [comment]
Show me on Tuesday

**Comment by collisdigital** [comment]
Having seen it on a slow connection, can we increase the submitting timer to 10 seconds instead of 5?

**Review by bennpearson — CHANGES_REQUESTED** [review]
Animation appears to stop after 5 seconds. Not sure if this is something that chrome dev tools throttling is introducing though

**Comment by ajmaddaford** [comment]
I've just tried this and the animation runs for 10 seconds. The button resizes when the text changes to "Submitting..." which we may not want?

File: app/assets/js/app/modules/submitter.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+      count++
+
+      trackEvent('send', {
+        hitType: 'event',
+        eventCategory: 'Error',
+        eventAction: 'Submission timeout',
+        eventLabel: `Attempts: ${count.toString()}`
+      })
+    }, 10000)
```
Comment (ajmaddaford) [inline app/assets/js/app/modules/submitter.js]: We should use a constant for this as it's a little unclear using the value directly.

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

File: app/assets/js/app/modules/submitter.js (hunk 1 above)
Comment (hamishtaplin) [reply app/assets/js/app/modules/submitter.js]: Do we want to make this configurable or just more explicit?

Comment (ajmaddaford) [reply app/assets/js/app/modules/submitter.js]: Just more explicit

**Review by bennpearson — APPROVED** [review]

**Review by ajmaddaford — APPROVED** [review]

**Comment by jonnyshaw89** [comment]
@hamishtaplin Can you update the description on this PR?

**Review by jonnyshaw89 — APPROVED** [review]

**Pull request merged**